shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveydetail/SurveyDetailViewModel.kt

```diff
@@ -86,4 +139,42 @@ class SurveyDetailViewModel(
             )
         }
     }
+
+    private fun setNextQuestionState() {
+        val questionSize = viewState.value.surveyDetail?.questions?.size ?: 0
+        val currentState = questionViewState.value
+        var nextQuestionIndex = currentState.currentQuestionIndex + 1
+        val isFinalQuestion = nextQuestionIndex >= (questionSize - 1)
+        val newState = SurveyQuestionViewState(
+            isFinalQuestion,
+            currentState.isLoading,
+            nextQuestionIndex
+        )
+        questionMutableViewState.update {
+            newState
+        }
+    }
+
+    private fun handleSubmitError() {
+        val currentState = questionViewState.value
+        val newState = SurveyQuestionViewState(
+            currentQuestionIndex = currentState.currentQuestionIndex
+        )
+        questionMutableViewState.update {
+            newState
+        }
```

Comment:
Since `currentState` is only used once, can we put it directly into the `newState` creation instead? 🙏
```suggestion
        questionMutableViewState.update {
            SurveyQuestionViewState(
                currentQuestionIndex = questionViewState.value.currentQuestionIndex
            )
        }
```